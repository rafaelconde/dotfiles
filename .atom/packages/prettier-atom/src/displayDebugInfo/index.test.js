jest.mock('read-pkg-up');
jest.mock('../atomInterface');
jest.mock('../helpers/getPrettierPath');

const readPkgUp = require('read-pkg-up');
const displayDebugInfo = require('./index');
const { getAtomVersion, getPrettierAtomConfig, addInfoNotification } = require('../atomInterface');
const { getGlobalPrettierPath } = require('../helpers/getPrettierPath');

test('it displays a notification on Atom with package information', () => {
  let title;
  let options;
  addInfoNotification.mockImplementation((_title, _options) => {
    title = _title;
    options = _options;
  });
  readPkgUp.sync.mockImplementation(() => ({
    pkg: {
      version: 'FAKE_PACKAGE_VERSION',
    },
  }));
  getAtomVersion.mockImplementation(() => 'FAKE_ATOM_VERSION');
  getPrettierAtomConfig.mockImplementation(() => 'FAKE_PRETTIER_ATOM_CONFIG');
  getGlobalPrettierPath.mockImplementation(() => 'FAKE_PRETTIER_PATH');

  displayDebugInfo();

  expect(title).toContain('prettier-atom');
  expect(options.detail).toMatchSnapshot();
  expect(options.dismissable).toBe(true);
});
