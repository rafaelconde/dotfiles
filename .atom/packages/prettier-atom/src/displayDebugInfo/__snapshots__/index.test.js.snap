// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays a notification on Atom with package information 1`] = `
"Atom version: FAKE_ATOM_VERSION
prettier-atom version: FAKE_PACKAGE_VERSION
prettier: FAKE_PRETTIER_PATH
prettier version: FAKE_PACKAGE_VERSION
prettier-eslint version: FAKE_PACKAGE_VERSION
prettier-atom configuration: \\"FAKE_PRETTIER_ATOM_CONFIG\\""
`;
